pageserver/src/tenant/ephemeral_file.rs

```diff
@@ -461,9 +511,17 @@ mod tests {
         test_read(cap - 10, 10).await;
         // read across file and buffer
         test_read(cap - 10, 20).await;
-        // stay from start of buffer
+        // stay from start of maybe flushed buffer
         test_read(cap, 10).await;
-        // completely within buffer
+        // completely within maybe flushed buffer
         test_read(cap + 10, 10).await;
+        // border onto edge of maybe flushed buffer.
+        test_read(cap * 2 - 10, 10).await;
+        // read across maybe flushed and mutable buffer
+        test_read(cap * 2 - 10, 20).await;
+        // read across three segments
+        test_read(cap - 10, cap + 20).await;
+        // completely within mutable buffer
+        test_read(cap * 2 + 10, 10).await;
```

Comment:
We should have 3 variants that execute the body of this test, really
1. flusher done
2. flusher in-progress
3. flusher hasn't started flushing yet

Reply:
invoke test for the three cases: 5acc61bdbcda38c93e6825da91a18706d3f1a7f3